src/modules/deal/deals.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+import { DEFAULTS } from '@ukef/constants/properties';
+import { AcbsService } from '@ukef/modules/acbs/acbs.service';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+
+import { CreateDealDto } from './dto/deals-request.dto';
+
+@Injectable()
+export class DealsService {
+  constructor(private readonly acbsAuthenticationService: AcbsAuthenticationService, private readonly acbsService: AcbsService) {}
+
+  async createDeal(createDealDto: CreateDealDto): Promise<any> {
+    const idToken = await this.acbsAuthenticationService.getIdToken();
```

Comment:
Can we please validate `idToken` is not void?

Reply:
I believe this error handling is already within `acbs-authentication.service.ts`, for example in `extractIdTokenFromConnectResponse`.
```
  private extractIdTokenFromConnectResponse(response: AxiosResponse<IdpConnectResponse>): string {
    const { id_token: idToken } = response.data || {};

    if (!idToken || typeof idToken !== 'string') {
      throw new AcbsAuthenticationFailedException('ID token was not returned by the IdP.');
    }
    return idToken;
  }
```

The other methods there will throw exceptions for various cases of failure to get a valid token.